db/attachment_compaction.go

```diff
@@ -15,12 +16,11 @@ import (
 
 const CompactionIDKey = "compactID"
 
-func Mark(db *Database, compactionID string, terminator chan struct{}, markedAttachmentCallback func(markedAttachments *int)) (int, error) {
+func Mark(db *Database, compactionID string, terminator chan struct{}, markedAttachmentCountPtr *uint64) (uint64, error) {
```

Comment:
On the blip side we're using SgwIntStat to encapsulate synchronized stat handling, even when those stats aren't wired up to the metrics endpoint.  What do you think about following that approach?  It would be nice to have a standard way to share stats between goroutines that we use across the product.

Reply:
I'm not sure if I'm a fan of that really. SgwIntStat is built around its use as a metric on the endpoint. Really you should initialize that with `NewIntStat` which has items we don't want in this case. I think it could be used by just initializing as `SgwIntStat{}` but I think that has the possibility to misuse it a little.
I accept that maybe passing *int into this function could be obscuring its use a little though, but I tried to minimize that in the parameter naming.

Reply:
Add an AtomicInt type, maybe embed in stats implementation.